src/reaktoro_pse/reaktoro_block.py

```diff
@@ -0,0 +1,952 @@
+                chemical_property_step=chemical_property_absolute_value*chemical_property_derivative*step
+            """,
+        ),
+    )
+    CONFIG.declare(
+        "jacobian_scaling_type",
+        ConfigValue(
+            default=JacScalingTypes.variable_scaling,
+            domain=IsInstance((str, JacScalingTypes)),
+            description="Defines how to scale Jacobian matrix",
+            doc="""
+            Defines methods for jacobian scaling:
+            - if option is no_scaling, jacobian scale will == 1 for all outputs
+            - if option is 'variable_scaling' will use output variable scaling factors
+            - if option is jacobian_matrix will use actual jac matrix to calculate scaling factors
+            - if user_scaling is not None then uses user provided scaling
+            """,
+        ),
+    )
+    CONFIG.declare(
+        "jacobian_user_scaling",
+        ConfigValue(
+            default=None,
+            domain=IsInstance((float, list, dict)),
+            description="Manual scaling factors for jacobian",
+            doc="""
+            Applies user provided jacobian scaling values:
+             - either single value that will be applied to all outputs in jacobian
+             - array applied across jacobian
+             - dict that specifics output and scaling factor to which apply scaling, (variable_scaling will be applied to non specified outputs)
+                e.g. {output_name:scaling_factor} applies to specific jac output 
+            """,
+        ),
+    )
+    CONFIG.declare(
+        "tolerance",
+        ConfigValue(
+            default=1e-8,
+            domain=float,
+            description="Tolerance for reaktoro solver",
+            doc="""Tolerance for primary reaktoro solver""",
+        ),
+    )
+    CONFIG.declare(
+        "epsilon",
+        ConfigValue(
+            default=1e-32,
+            domain=float,
+            description="Epsilon for reaktoro solver",
+            doc="""Defines what is considered to be 0 for ion composition""",
+        ),
+    )
+    CONFIG.declare(
+        "max_iters",
+        ConfigValue(
+            default=400,
+            domain=int,
+            description="Maximum number of iterations for reaktoro solver",
+            doc="""The maximum number of iterations for reaktoro solver""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_during_initialization",
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description="Option to pre-solve to low tolerance first, before primary solve but only during initialization",
+            doc="""In some cases reaktoro might fail to solve to high tolerance first,
+            a presolve at low tolerance can enable the reaktoro solve to high tolerance, this will only presolve during initialization""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_property_block",
```

Comment:
This might not be a good idea, but to put an idea out there for consideration should we use nested `ConfigBlocks` here? I.e.

```
PresolveConfig = ConfigDict()
PresolveConfig.declare("presolve_tolerance", ConfigValue(...))

CONFIG.decalre("presolve", PresolveConfig)
```

This might help break up the code for us for maintenance (e.g. easier to find the presolve config arguments) and might help the user too by separating the critical arguments from optional sub-sets.

Reply:
It would also help keep the different tolerances more separate.

Reply:
yes - thats a good idea. .. I will see if I have time to do this refactor before first merge.